### PR TITLE
Nmr 4639 sharepoint api

### PR DESCRIPTION
**Pull request by vfiack**

### CODE REVIEW

File: src/main/java/com/nanalysis/sharepoint/SharepointClient.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                .POST(HttpRequest.BodyPublishers.ofString(authentication))
+                .build();
+
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        return extractXmlTag(response.body(), "BinarySecurityToken");
+    }
+
+    private void signin(String securityToken) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(baseUrl + "/_forms/default.aspx?wa=wsignin1.0"))
```
Comment (julienpontabry) [inline src/main/java/com/nanalysis/sharepoint/SharepointClient.java:86]: That smells the URL that may change without notice 🤪

File: src/main/java/com/nanalysis/sharepoint/Templates.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            "          <a:Address>${endpoint}</a:Address>\n" +
+            "        </a:EndpointReference>\n" +
+            "      </wsp:AppliesTo>\n" +
+            "      <t:KeyType>http://schemas.xmlsoap.org/ws/2005/05/identity/NoProofKey</t:KeyType>\n" +
+            "      <t:RequestType>http://schemas.xmlsoap.org/ws/2005/02/trust/Issue</t:RequestType>\n" +
+            "      <t:TokenType>urn:oasis:names:tc:SAML:1.0:assertion</t:TokenType>\n" +
+            "    </t:RequestSecurityToken>\n" +
+            "  </s:Body>\n" +
+            "</s:Envelope>";
+}
```
Comment (julienpontabry) [inline src/main/java/com/nanalysis/sharepoint/Templates.java:53]: That looks obvious 😵‍💫

File: src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+module com.nananlysis.jcamp {
+    requires org.json;
+    requires java.xml;
+    requires java.net.http;
+    exports com.nanalysis.sharepoint;
+}
```
Comment (julienpontabry) [inline src/main/java/module-info.java:21]: I admire your courage for using java modules !

**Review by julienpontabry — COMMENTED** [review]
The code looks good to me (although I am not an expert in such programming).

File: src/main/java/com/nanalysis/sharepoint/SharepointClient.java (hunk 1 above)
Comment (vfiack) [reply src/main/java/com/nanalysis/sharepoint/SharepointClient.java:86]: The endpoint to get the security token looks even worse. I didn't find the relevant documentation, you only find that in forums and stackoverflow. Not sure where it comes from in the first place.

**Review by julienpontabry — APPROVED** [review]

**Pull request merged**